### PR TITLE
Send shutdown messages to users when signaltower shuts down

### LINKED ISSUE #20: Send shutdown event when shutting down
When the signal tower is shut down, according to the palava protocol a shutdown event shall be sent:
```
{
  event: 'shutdown'
}
```

### PR DESCRIPTION
**Pull request by farao**
Closes #20

### CODE REVIEW

File: lib/signal_tower.ex [hunk 1]
```diff
@@ -45,4 +40,16 @@ defmodule SignalTower do
...
+  def stop(_) do
+    DynamicSupervisor.which_children(Room.Supervisor)
+    |> Enum.map(fn
+      {:undefined, pid, :worker, _} when is_pid(pid) ->
+        send(pid, :shutdown)
+
+      _ ->
+        :ok
+    end)
+  end
```
Comment (erikzenker) [inline lib/signal_tower.ex:54]: Is the signal handler?

Comment (erikzenker) [reply lib/signal_tower.ex:54]: + this

Comment (erikzenker) [reply lib/signal_tower.ex:54]: ... Is this the signal handler

File: lib/signal_tower.ex [hunk 2]
```diff
@@ -45,4 +40,16 @@ defmodule SignalTower do
     Logger.info("SignalTower started")
     ret
   end
+
+  @impl Application
+  def stop(_) do
```
Comment (erikzenker) [inline lib/signal_tower.ex:45]: I liked the `_state` in the previous version, because it gives me an idea about the variable content/type

Comment (farao) [reply lib/signal_tower.ex:45]: Ok, I will bring it back ;-)

File: lib/signal_tower.ex (hunk 1 above)
Comment (farao) [reply lib/signal_tower.ex:54]: What do you mean by signal handler? stop(state) is a callback of the Application behaviour ("interface") and it gets called when the application is asked to shut down - is that what you mean?

Comment (erikzenker) [reply lib/signal_tower.ex:54]: Thats what i mean. I was refering to signal handlers used in c++ programms.

Comment (erikzenker) [reply lib/signal_tower.ex:54]: Is the 'asked to shutdown' compareable to unix signal e.g.: sigterm?

Comment (farao) [reply lib/signal_tower.ex:54]: Exactly. "By default, a SIGTERM from the operating system will automatically translate to System.stop/0." https://hexdocs.pm/elixir/Application.html

Comment (erikzenker) [reply lib/signal_tower.ex:54]: Thx a lot for clarification!

**Pull request merged**